feat/geozarr model

d-v-b opened the pull request:
initial working pydantic models for geozarr

d-v-b commented on src/eopf_geozarr/data_api/geozarr/common.py:
```diff
@@ -0,0 +1,161 @@
+CF_STANDARD_NAMES = get_cf_standard_names(url=CF_STANDARD_NAME_URL)
+
+
+def check_standard_name(name: str) -> str:
```
Pydantic does most of its validation routines based on type annotations. When we annotate an attribute on a pydantic model with this type: https://github.com/d-v-b/data-model/blob/3d11af412e460993f8e603dcff0555c5342c4e8f/src/eopf_geozarr/data_api/geozarr/common.py#L70, then pydantic will run the `check_standard_name` function after checking that the input is a string.

emmanuelmathot commented:
@d-v-b What is the status of this PR?

d-v-b commented:
@emmanuelmathot I am getting an interesting test failure that would be helpful for you to check. Within a dataset, for each data variable that declares coordinates (`a`, `b`, `c`) via `{"attributes": {"coordinates": "a b c"}}`, `a`, `b`, and `c` must all be the names of coordinate variable arrays in the dataset. But in my test data (which I generated via the conversion script and is included as a [test fixture](https://github.com/d-v-b/data-model/blob/9a79771e171c8a3824e2b11f7c188f392809144d/tests/test_data_api/conftest.py#L6) in this PR), I'm finding some data variables (such as `"quality/l1c_quicklook/r10m/5/tci"`) that declare a coordinate "band" that's not present as a coordinate variable. Is this consistent with the data model?

emmanuelmathot commented:
@d-v-b As a more general question. Is this `{"attributes": {"coordinates": "a b c"}}` declaration a convention or a specifications of any sort? If no, why is it there, Shouldn't this be actually declared in a proper way in addition of fixing the coordinates existence check?

emmanuelmathot commented:
2 related issues created following the `band` issue

- https://github.com/pydata/xarray/pull/10720
- https://gitlab.eopf.copernicus.eu/cpm/eopf-cpm/-/issues/836

d-v-b commented:
I added a new `types` module that just contains types to replace the use of `Dict[str, Any]` that occur throughout `geozarr.py`, and updated the type annotations accordingly. This might be out of scope, in which case I'm happy to spin this out into a new PR.
